src/integrationTest/java/org/opensearch/security/SearchOperationTest.java

```diff
@@ -1344,5 +1607,18 @@ public void shouldRestoreSnapshot_failureOperationForbidden() throws IOException
 			assertThat(internalClient, not(clusterContainsDocument(RESTORED_SONG_INDEX_NAME, "Eins")));
 			assertThat(internalClient, not(clusterContainsDocument(RESTORED_SONG_INDEX_NAME, "Zwei")));
 		}
+		auditLogsRule.assertExactlyOne(userAuthenticated(LIMITED_WRITE_USER).withRestRequest(PUT, "/_snapshot/test-snapshot-repository"));
+		auditLogsRule.assertExactlyOne(userAuthenticated(LIMITED_WRITE_USER).withRestRequest(PUT, "/_snapshot/test-snapshot-repository/restore-snapshot-negative-forbidden-operation"));
+		auditLogsRule.assertExactlyOne(userAuthenticated(LIMITED_READ_USER).withRestRequest(POST, "/_snapshot/test-snapshot-repository/restore-snapshot-negative-forbidden-operation/_restore"));
+		auditLogsRule.assertExactlyOne(userAuthenticated(LIMITED_WRITE_USER).withRestRequest(POST, "/_bulk"));
+		auditLogsRule.assertAtLeast(1, userAuthenticated(LIMITED_WRITE_USER).withRestRequest(GET, "/_snapshot/test-snapshot-repository/restore-snapshot-negative-forbidden-operation"));
```

Comment:
Why do we need AtLeast, shouldn't this be consistent?

Reply:
(Multiple places)

Reply:
I noticed this when I added assertions for tests related to snapshots. It depends on the test execution order. For example when the snapshot is created the first time then the following audit logs are created:

1. Two: `userAuthenticated(LIMITED_WRITE_USER).withRestRequest(GET, "/_snapshot/test-snapshot-repository/snapshot-positive-test")`
2. Four `grantedPrivilege(LIMITED_WRITE_USER, "GetSnapshotsRequest")`

But each subsequent test which creates snapshots generates distinct audit records

1. Only one: `userAuthenticated(LIMITED_WRITE_USER).withRestRequest(GET, "/_snapshot/test-snapshot-repository/snapshot-positive-test")` instead of two
2. Only two `grantedPrivilege(LIMITED_WRITE_USER, "GetSnapshotsRequest")` instead of four

I am not sure if this is expected behaviour.

Reply:
That is really strange behavior, would you be willing to write up a bug on it?  It looks like we've got an easy way to reproduce with these tests.  If we have the tracking issue, I'm fine keeping the atLeast functionality to keep the scenario tested while we narrow in on the source of the behavior.

Reply:
I created a bug: https://github.com/opensearch-project/security/issues/2169